build/common/installer/scripts/tomlparser-hostlogs-geneva-config.rb

```diff
@@ -0,0 +1,145 @@
+          if GENEVA_SUPPORTED_ENVIRONMENTS.include?(geneva_account_environment)
+            @geneva_account_environment = geneva_account_environment
+            @geneva_account_name = geneva_account_name
+            @geneva_account_namespace = geneva_account_namespace
+            @geneva_logs_config_version = geneva_logs_config_version
+          else
+            puts "config::error:unsupported geneva config environment"
+          end
+        else
+          puts "config::error:invalid geneva logs config"
+        end
+        geneva_gcs_authid = geneva_logs_config[:authid]
+        if geneva_gcs_authid.nil? || geneva_gcs_authid.empty?
+          # extract authid from nodes config
+          begin
+            file = File.read(@azure_json_path)
+            data_hash = JSON.parse(file)
+            sp_client_id = data_hash["aadClientId"]
+            user_assigned_client_id = data_hash["userAssignedIdentityID"]
+            if (!sp_client_id.nil? &&
+                !sp_client_id.empty? &&
+                sp_client_id.downcase == "msi" &&
+                !user_assigned_client_id.nil? &&
+                !user_assigned_client_id.empty?)
+              geneva_gcs_authid = "client_id##{user_assigned_client_id}"
+              puts "using authid for geneva integration: #{geneva_gcs_authid}"
+              @geneva_gcs_authid = geneva_gcs_authid
+            end
+          rescue => errorStr
+            puts "failed to get user assigned client id with an error: #{errorStr}"
+          end
+        elsif !geneva_gcs_authid.start_with?("client_id#") && !geneva_gcs_authid.start_with?("object_id#") && !geneva_gcs_authid.start_with?("mi_res_id#")
+            puts "config::error:auth id must be in one of the suppported formats: object_id#<guid> or client_id#<guid> or mi_res_id#<identity resource id>"
+        else
+          @geneva_gcs_authid = geneva_gcs_authid
+        end
+      end
+    end
+  rescue => errorStr
+    puts "config::error:Exception while reading config settings for agent configuration setting - #{errorStr}, using defaults"
+  end
+end
+
+# Write the settings to file, so that they can be set as environment variables
+def writeEnvScript(filepath)
+  file = File.open(filepath, "w")
+
+  if !file.nil?
+
+    if !@geneva_account_environment.empty? && !@geneva_account_name.empty? && !@geneva_account_namespace.empty? && !@geneva_logs_config_version.empty? && !@geneva_gcs_authid.empty?
+      file.write(get_command_windows("MONITORING_GCS_ENVIRONMENT", @geneva_account_environment))
+      file.write(get_command_windows("MONITORING_GCS_ACCOUNT", @geneva_account_name))
+      file.write(get_command_windows("MONITORING_GCS_NAMESPACE", @geneva_account_namespace))
+      file.write(get_command_windows("MONITORING_CONFIG_VERSION", @geneva_logs_config_version))
+      
+      authIdParts =  @geneva_gcs_authid.split('#', 2)
```

Comment:
if the @geneva_gcs_authid is not valid, what will be behavior? is container gets crashed?

Reply:
The monitoring agent just crashes and will retry connecting again every 120 seconds. The container however is seen as healthy in this instance.

Reply:
log the error message if the authid is not valid format.

Reply:
Line 85 - 92 has logging for invalid authids. If there is a condition, we are overlooking let us know.